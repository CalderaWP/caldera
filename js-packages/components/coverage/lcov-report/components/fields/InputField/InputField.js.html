<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/fields/InputField/InputField.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/fields/InputField</a> InputField.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {parseAttributes,fieldClassNames,isValidHtml5type,toBoolean} from '../util';
&nbsp;
import PropTypes from 'prop-types';
import React from 'react'
import {TextControl,CheckboxControl} from "@wordpress/components";
export const InputField = ({
							  label,
							  description,
							  fieldId,
							  placeholder,
							  required,
							  html5type,
							  value,
							  onChange,
							  onBlur,
							  attributes,
	children
						  }) =&gt; {
&nbsp;
	const fieldType = isValidHtml5type(html5type) ? html5type : 'text';
	const _attributes = parseAttributes(attributes,fieldType);
	if ('checkbox' === fieldType) {
&nbsp;
		const checked = value ? true : false;
		return (
			&lt;CheckboxControl
				id={fieldId}
				checked={checked}
				help={description}
				{..._attributes}
				onChange={ onChange }
				label={label}
			/&gt;
		)
&nbsp;
	} else {
		return (
			&lt;TextControl
				label={label}
				className={fieldClassNames(fieldType)}
				id={fieldId}
				value={value}
				placeholder={placeholder}
				type={fieldType}
				onChange={onChange}
				onBlur={onBlur}
				help={description}
				{..._attributes}
			/&gt;
		)
	}
};
&nbsp;
InputField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	html5type: PropTypes.string,
	attributes: PropTypes.object,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.array,
		PropTypes.bool
	]),
	onChange: PropTypes.func,
	onBlur: PropTypes.func
};
&nbsp;
InputField.defaultProps = {
	onBlur: <span class="fstat-no" title="function not covered" >()</span> =&gt; {
	},
	required: false,
	html5type: 'text'
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 31 2018 13:11:46 GMT-0500 (EST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
