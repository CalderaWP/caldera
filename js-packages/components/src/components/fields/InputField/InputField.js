import {parseAttributes,fieldClassNames,isValidHtml5type} from '../util';
import PropTypes from 'prop-types';
import React from 'react'

import {FieldLabel} from '../FieldLabel/FieldLabel'
import {FieldWrapper} from '../FieldWrapper/FieldWrapper'
import {Input} from '../controls/Input';
export const InputField = ({
							  label,
							  description,
							  fieldId,
							  placeholder,
							  required,
							  html5type,
							  value,
							  onChange,
							  onBlur,
							  attributes,
	children
						  }) => {

	const fieldType = isValidHtml5type(html5type) ? html5type : 'text';
	const _attributes = parseAttributes(attributes,fieldType);
	return (
		<FieldWrapper
			fieldType={fieldType}
		>
			<FieldLabel
				fieldId={fieldId}
			>
				{label}
			</FieldLabel>
			<Input
				fieldId={fieldId}
				value={value}
				placeholder={placeholder}
				type={fieldType}
				onChange={onChange}
				onBlur={onBlur}
				help={description}
				attributes={_attributes}
			/>
			{children}
		</FieldWrapper>
	)
};

InputField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	placeholder: PropTypes.string,
	required: PropTypes.bool,
	html5type: PropTypes.string,
	attributes: PropTypes.object,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.array
	]),
	onChange: PropTypes.func,
	onBlur: PropTypes.func
};

InputField.defaultProps = {
	onBlur: () => {
	},
	required: false,
	html5type: 'text'
};
