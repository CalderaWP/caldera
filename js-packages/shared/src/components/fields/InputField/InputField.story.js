import React from 'react';
import { storiesOf } from '@storybook/react';
import { InputField } from './InputField';

const onChange = (event) => {

};
const onBlur = (event) => {

};
storiesOf('InputField', module).add('With no value', () => (
	<InputField
		label={'Hi Roy'}
		html5type={'text'}
		onChange={onChange}
		onBlur={onBlur}
	/>
));
