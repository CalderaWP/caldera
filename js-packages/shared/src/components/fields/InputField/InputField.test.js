import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { InputField } from './InputField';

describe('InputField ', () => {
	let onChange;
	let onBlur;

	beforeEach( () => {
		onChange= jest.fn();
		onBlur= jest.fn()
	});

	it( 'matches snapshot with all props', () =>  {
		const component = renderer.create(<InputField
			label={'Hi Roy'}
			description={'Say Hi'}
			placeholder={'Hello'}
			html5type={'text'}
			value={'Roy'}
			onChange={onChange}
			onBlur={onBlur}
		/>);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Uses text when html5type is not valid', () =>  {
		const component = mount(<InputField
			label={'Hi Roy'}
			description={'Say Hi'}
			placeholder={'Hello'}
			html5type={'unreal'}
			value={'Roy'}
			onChange={onChange}
			onBlur={onBlur}
		/>);

		expect( component.find( 'input').prop( 'type') ).toEqual( 'text' );
	});
});
