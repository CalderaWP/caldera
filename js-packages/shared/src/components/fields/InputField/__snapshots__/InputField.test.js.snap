// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField  matches snapshot with all props 1`] = `
<div
  className="caldera-field caldera-field-text"
>
  <label
    className="caldera-field-label caldera-field-label-undefined"
    htmlFor={undefined}
  >
    Hi Roy
  </label>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby="inspector-text-control-0__help"
        className="components-text-control__input"
        id={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Hello"
        type="text"
        value="Roy"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-0__help"
    >
      Say Hi
    </p>
  </div>
</div>
`;
